components/brave_wallet/common/zcash_utils.cc

```diff
@@ -5,18 +5,93 @@
 
 #include "brave/components/brave_wallet/common/zcash_utils.h"
 
-#include <optional>
+#include <algorithm>
+#include <string>
+#include <utility>
+#include <vector>
 
+#include "base/big_endian.h"
+#include "base/types/expected.h"
 #include "brave/components/brave_wallet/common/btc_like_serializer_stream.h"
 #include "brave/components/brave_wallet/common/encoding_utils.h"
+#include "brave/components/brave_wallet/common/f4_jumble.h"
 #include "brave/components/brave_wallet/common/hash_utils.h"
 #include "brave/third_party/bitcoin-core/src/src/base58.h"
+#include "brave/third_party/bitcoin-core/src/src/bech32.h"
+#include "brave/third_party/bitcoin-core/src/src/util/strencodings.h"
+#include "components/base32/base32.h"
 
 namespace brave_wallet {
 
 namespace {
+constexpr size_t kPaddedHrpSize = 16;
 constexpr size_t kPubKeyHashSize = 20;
 constexpr size_t kPrefixSize = 2;
+
+// https://zips.z.cash/zip-0316#encoding-of-unified-addresses
+enum AddrType {
+  P2PKH = 0x00,
+  P2PSH = 0x01,
+  Sapling = 0x02,
+  Orchard = 0x03,
+  Last = 0x04
```

Comment:
This is typically done as `kMaxValue = Orchard`, and compared as `type > AddrType::kMaxValue`
https://source.chromium.org/search?q=%22kMaxValue%20%3D%22&ss=chromium&start=41
Also consider `k` prefix for enumerations

Reply:
Done